### PR TITLE
chore: bump oxc to v0.17.1

### PR DESCRIPTION
**Pull request by Boshen**
* Preserves parentheses from the original code
* simple constant folding and dead code removal

### CODE REVIEW

File: crates/rolldown/tests/esbuild/dce/no_side_effects_comment/artifacts.snap [hunk 1]
```diff
@@ -10,7 +10,7 @@ input_file: crates/rolldown/tests/esbuild/dce/no_side_effects_comment
 ```js
 
 //#region expr-arrow.js
-x([/* #__NO_SIDE_EFFECTS__ */ (y) => y, /* #__NO_SIDE_EFFECTS__ */ () => {}, /* #__NO_SIDE_EFFECTS__ */ (y) => y, /* #__NO_SIDE_EFFECTS__ */ async (y) => y, /* #__NO_SIDE_EFFECTS__ */ async () => {}, /* #__NO_SIDE_EFFECTS__ */ async (y) => y,]);
+x([/* #__NO_SIDE_EFFECTS__ */ (y) => y, /* #__NO_SIDE_EFFECTS__ */ () => {}, /* #__NO_SIDE_EFFECTS__ */ (y) => (y), /* #__NO_SIDE_EFFECTS__ */ async (y) => y, /* #__NO_SIDE_EFFECTS__ */ async () => {}, /* #__NO_SIDE_EFFECTS__ */ async (y) => (y),]);
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/dce/no_side_effects_comment/artifacts.snap:13]: This is correct, but not necessary, let's just merge it for now.

**Review by IWANABETHATGUY — DISMISSED** [review]

**Comment by Boshen** [comment]
```
export { a }
         ^ this became identifier reference instead of export name
```

**Comment by Boshen** [comment]
Reason for broken tests: https://github.com/oxc-project/oxc/issues/4060
Method for prevent future releases affecting Rolldown: https://github.com/oxc-project/monitor-oxc/issues/7

File: Cargo.toml [hunk 2]
```diff
@@ -138,8 +138,8 @@ urlencoding        = "2.1.3"
 vfs                = "0.12.0"
 xxhash-rust        = "0.8.10"
 # oxc crates share the same version
-oxc                = { version = "0.16.2", features = ["sourcemap_concurrent", "transformer", "minifier"] }
```
Comment (Boshen) [inline Cargo.toml:141]: Testing from main because I don't know what's going on in my local test.

**Comment by Boshen** [comment]
Teach me how to debug this one 

"rollup@function@allow-undefined-as-parameter: do not make assumptions about its value when using undefined as a parameter name"

**Comment by Boshen** [comment]
I think moving forwards we should enable all tests and record all the passed and failed ones instead of needing to manually turn on and off tests.

**Comment by hyf0** [comment]
> Teach me how to debug this one

> "rollup@function@allow-undefined-as-parameter: do not make assumptions about its value when using undefined as a parameter name"

add `solo: true` in https://github.com/rollup/rollup/blob/64870c31062983a741832d8cd2d4bcdf55e4de55/test/function/samples/allow-undefined-as-parameter.

> I think moving forwards we should enable all tests and record all the passed and failed ones instead of needing to manually turn on and off tests.

Already have one: https://github.com/rolldown/rolldown/blob/main/packages/rollup-tests/src/status.md.

**Comment by Boshen** [comment]
Can merge after tests pass.

**Comment by hyf0** [comment]
Nice!

**Review by hyf0 — APPROVED** [review]

**Pull request merged**